### PR TITLE
Alerting: Add custom templated title to Wecom notifier

### LINKED ISSUE #51528: Wecom contact channel can't define the title which I want
**What happened**:
1. I add a wecom contact point  and notification policy , then click test msg, then I got a message in wecom.
image
2. I add a wecom contact point  and notification policy , then add a rule for example cpu usage > 10% ,then I got a alert message in wecom.
image

**What you expected to happen**:
I can define the msg title which I want in test and alert msg  ,like this message 
image

**How to reproduce it (as minimally and precisely as possible)**:
   Use wecom contact point as notification type ,and you will reproduce it .
**Anything else we need to know?**:

**Environment**:
- Grafana version: docker image grafana/grafana:9.0
- Data source type & version: Prometheus 2.26.0
- OS Grafana is installed on: ubuntu18.04 and docker 20.10.17
- User OS & Browser: 
- Grafana plugins:
- Others:
**Link Issues**
#48159

### PR DESCRIPTION
**Pull request by dingweiqings**
**What this PR does / why we need it**:
When someone need to change  title in wecom  alert message , and this pr  support this case.

**Which issue(s) this PR fixes**:
#51528 #48159

**Special notes for your reviewer**:
I follow the code pattern about custom email subject .

### CODE REVIEW

**Comment by JacobsonMT** [comment]
Hi @dingweiqings, thanks for the contribution! This is looking good but is missing a couple pieces, namely:

1. This new `Title` field needs to be confiigurable in the Grafana UI. This can be done by adding a new field in `pkg/services/ngalert/notifier/available_channels.go`. An example can be seen here: https://github.com/grafana/grafana/pull/49742/files#diff-1a7dd1bf6b0704888c591e444fe45e45dabcf01d11d744f298fa17e18b4238d1.
2. A unit test should be added in `wecom_test.go` that tests templating this new field. An example can be seen here: https://github.com/grafana/grafana/pull/49742/files#diff-bc26976b3571e721b0327caaa80d1300fe6775a8106a232ce98063af42764a01.

Let me know if you need any help with either of these, I'll be glad to give a hand.

**Comment by dingweiqings** [comment]
@JacobsonMT Do I  need to do something else work  ?

File: pkg/services/ngalert/notifier/available_channels.go [hunk 1]
```diff
@@ -696,6 +696,14 @@ func GetAvailableNotifiers() []*alerting.NotifierPlugin {
 					Placeholder:  `{{ template "default.message" . }}`,
 					PropertyName: "message",
 				},
+				{ // New in 9.0.
```
Comment (JacobsonMT) [inline pkg/services/ngalert/notifier/available_channels.go:699]:
```suggestion
				{ // New in 9.1.
```

NIT: Wrong version in the comment

File: pkg/services/ngalert/notifier/channels/wecom_test.go [hunk 2]
```diff
@@ -76,6 +76,33 @@ func TestWeComNotifier(t *testing.T) {
 				"msgtype": "markdown",
 			},
 			expMsgError: nil,
+		}, {
+			name: "Custom title and message with multiple alerts",
+			settings: `{
+				"url": "https://qyapi.weixin.qq.com/cgi-bin/webhook/send?key=db0e4533-aa2a-4385-8a24-2a7b6be72d20",
```
Comment (JacobsonMT) [inline pkg/services/ngalert/notifier/channels/wecom_test.go:82]:
```suggestion
				"url": "http://localhost",
```
Let's use localhost instead a "real" url here.

**Review by JacobsonMT — CHANGES_REQUESTED** [review]
Looks great! Should be good to approve/merge after addressing the one small issues I mentioned.

File: pkg/services/ngalert/notifier/available_channels.go [hunk 3]
```diff
@@ -696,6 +696,14 @@ func GetAvailableNotifiers() []*alerting.NotifierPlugin {
 					Placeholder:  `{{ template "default.message" . }}`,
 					PropertyName: "message",
 				},
+				{ // New in 9.1.
+					Label:        "Title",
+					Element:      alerting.ElementTypeInput,
+					InputType:    alerting.InputTypeText,
+					Description:  "Templated title of the email",
```
Comment (JacobsonMT) [inline pkg/services/ngalert/notifier/available_channels.go:703]: Ah, one last thing, the description mentions email.

```suggestion
					Description:  "Templated title of the message",
```

**Review by JacobsonMT — APPROVED** [review]
LGTM! Thanks again for the contribution :rocket:

**Pull request merged**